### ISSUE TITLE
`no-await-in-promise-methods`: Report `await` nested inside other expressions

### ISSUE THREAD

**Issue by abrahamguo** [sindresorhus/eslint-plugin-unicorn#3753]
### Description

An `await` with other operations following it has the same type of problem as the cases that are already reported by this rule, so it should report them.

### Examples

```js
// ❌
Promise.all([await promise, anotherPromise]);

// ✅
Promise.all([Math.random() ? await promise : anotherPromise, anotherPromise]);
```

```js
// ❌
Promise.all([await promise, anotherPromise]);

// ✅
Promise.all([(await promise).foo, anotherPromise]);
```
